src/compiler/transformers/module/system.ts

```diff
@@ -23,8 +23,15 @@ namespace ts {
         const previousExpressionSubstitution = context.expressionSubstitution;
         context.enableExpressionSubstitution(SyntaxKind.Identifier);
         context.enableExpressionSubstitution(SyntaxKind.BinaryExpression);
+        context.enableExpressionSubstitution(SyntaxKind.PrefixUnaryExpression);
         context.enableExpressionSubstitution(SyntaxKind.PostfixUnaryExpression);
         context.expressionSubstitution = substituteExpression;
+        
+        context.enableEmitNotification(SyntaxKind.SourceFile);
+
+        const exportFunctionForFileMap: Identifier[] = [];

```

Comment:
Since we aren't guaranteed that the node ids for SourceFiles are sequential, we will end up with a sparse array. Would it be more efficient to use `Map<Identifier> = { }`?